### PR TITLE
add comment for ReadOptions.set_iterate_upper_bound method.

### PR DESCRIPTION
**Pull request by xiaobogaga**
the method `ReadOptions.set_iterate_upper_bound` method seems confusing, because the upper_bound is not returned when doing iteration.  some comment are added for this method.

### CODE REVIEW

**Comment by xiaobogaga** [comment]
@iSynaptic could you please take a look, thanks.

File: src/db.rs [hunk 1]
```diff
@@ -1768,9 +1768,10 @@ impl ReadOptions {
         }
     }
 
+    /// set_iterate_upper_bound sets the upper bound for an iterator, and the upper bound itself
```
Comment (aleksuss) [inline src/db.rs]: IMO `set_iterate_upper_bound` is redundant here, isn't it ?

Comment (iSynaptic) [reply src/db.rs]: Yes, I would say it is.

Comment (xiaobogaga) [reply src/db.rs]: thanks, I would fix it.

**Comment by xiaobogaga** [comment]
@aleksuss PTAL again, thanks.

**Pull request merged**